### PR TITLE
feat: Select current window and set size

### PR DESCRIPTION
**Pull request by Squidonomics**
Implements the `/session/:sessionId/window` and `/session/:sessionId/window/:windowHandle/size` endpoints.

### CODE REVIEW

File: Sources/WebDriver/Session.swift [hunk 1]
```diff
@@ -292,6 +301,20 @@ public class Session {
         try webDriver.send(Requests.SessionKeys(session: id, value: value))
     }
 
+    /// Change focus to another window
+    /// - Parameter name: The window to change focus to
+    public func window(name: String) throws {
```
Comment (compnerd) [inline Sources/WebDriver/Session.swift:306]: Does `focus(window:)` make more sense for this operation?

File: Sources/WebDriver/Session.swift [hunk 2]
```diff
@@ -292,6 +301,20 @@ public class Session {
...
+        try webDriver.send(Requests.SessionWindow(session: id, name: name))
+    }
+
+    /// Change the size of the specified window
+    /// - Parameter windowHandle: URL parameter is "current", the currently active window will be resized.
+    /// - Parameter width: The new window width.
+    /// - Parameter height: The new window height
+    public func windowHandleSize(windowHandle: String, width: Int, height: Int) throws {
```
Comment (compnerd) [inline Sources/WebDriver/Session.swift:314]: Does `resize(window:to:)` make more sense for this operation?

File: Sources/WebDriver/Session.swift (hunk 1 above)
Comment (Squidonomics) [reply Sources/WebDriver/Session.swift:306]: I like these names better I'll push up a change

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
This might require a rebase after #125 is merged so we can get pre-commit CI.

File: Docs/SupportedAPIs.md [hunk 3]
```diff
@@ -57,8 +57,8 @@ Contributions to expand support to unimplemented functionality are always welcom
 | POST   | `/session/:sessionId/touch/up`                      | Supported    | `Session.touchUp()` |
 | GET    | `/session/:sessionId/url`                           | Not supported| `Session.url`       |
 | POST   | `/session/:sessionId/url`                           | Not supported| `Session.url()`     |
-| DELETE | `/session/:sessionId/window`                        | Supported    | Not implemented     |
-| POST   | `/session/:sessionId/window`                        | Supported    | Not implemented     |
+| DELETE | `/session/:sessionId/window`                        | Supported    | `Session.window()`  |
+| POST   | `/session/:sessionId/window`                        | Supported    | `Session.window()`  |
```
Comment (jeffdav) [inline Docs/SupportedAPIs.md:61]: The swift-webdriver API column isn't quite right for these. 

I don't think there is a method for the `DELETE` version (though we should add one), and the `POST` version should be `Session.focus()`.

File: Sources/WebDriver/Session.swift [hunk 4]
```diff
@@ -292,6 +301,20 @@ public class Session {
         try webDriver.send(Requests.SessionKeys(session: id, value: value))
     }
 
+    /// Change focus to another window
+    /// - Parameter window: The window to change focus to
+    public func focus(window: String) throws {
+        try webDriver.send(Requests.SessionWindow(session: id, name: name))
+    }
```
Comment (jeffdav) [inline Sources/WebDriver/Session.swift:308]: Does this work?  The parameter `window` isn't used, and `name` isn't defined?

File: Sources/WebDriver/Session.swift [hunk 5]
```diff
@@ -292,6 +301,20 @@ public class Session {
...
+
+    /// Change the size of the specified window
+    /// - Parameter windowHandle: URL parameter is "current", the currently active window will be resized.
+    /// - Parameter width: The new window width.
+    /// - Parameter height: The new window height
+    public func resize(window: String, width: Int, height: Int) throws {
+        try webDriver.send(Requests.SessionWindowHandleSize(session: id, windowHandle: windowHandle, width: width, height: height))
+    }
```
Comment (jeffdav) [inline Sources/WebDriver/Session.swift:316]: Same as above, I think there's a mismatch between `window` and `windowHandle`?

File: Docs/SupportedAPIs.md (hunk 3 above)
Comment (jeffdav) [reply Docs/SupportedAPIs.md:61]: We should also update the rows for `/session/:sessionId/window/:windowHandle/size` below.

**Review by jeffdav — CHANGES_REQUESTED** [review]
This is really great!  Thanks for contributing.

I had a couple of minor comments, and additionally I think we should go ahead and add `func Session.close()` for `DELETE /session/:sessionId/window` and add `var size` for `GET /session/:sessionId/window/:windowHandle/size`, since we have the the requests defined already.

Does that make sense?  LMK if you have any questions.

File: Sources/WebDriver/Requests.swift [hunk 6]
```diff
@@ -462,6 +462,70 @@ public enum Requests {
...
+            public var body: Body { .init(name: name) }
+
+            public struct Body: Codable {
+                public var name: String
+            }
+        }
+    }
+
+    // https://www.selenium.dev/documentation/legacy/json_wire_protocol/#sessionsessionidwindowwindowhandlesize
+    public enum SessionWindowHandleSize {
```
Comment (compnerd) [inline Sources/WebDriver/Requests.swift:495]: On further reflection, I wonder if we should name this the `WindowSize` request?  Each request is implicitly tied to the session.  The URL format is `/session/:sid/window/:wid/size/:sz`.  So, `SessionWindowSize` or `WindowSize` makes more sense IMO.

Comment (jeffdav) [reply Sources/WebDriver/Requests.swift:495]: Ah yeah, good catch.

Comment (Squidonomics) [reply Sources/WebDriver/Requests.swift:495]: I might be looking at the wrong docs but the URL format looks to be `/session/:sessionId/window/:windowHandle/size`

Comment (jeffdav) [reply Sources/WebDriver/Requests.swift]: You're correct-- @compnerd was trying to save typing (and size goes in the body, not the URL, as you've coded it).

Comment (Squidonomics) [reply Sources/WebDriver/Requests.swift]: Size doesn't appear to be a passable parameter in either the URL or the body. You only post the width and height in the json body and calling `GET` will send those dimensions back. I could be misunderstanding something though.

**Comment by Squidonomics** [comment]
Is this ready to be merged @jeffdav ?

**Review by jeffdav — APPROVED** [review]

File: Tests/UnitTests/APIToRequestMappingTests.swift [hunk 7]
```diff
@@ -169,4 +169,26 @@ class APIToRequestMappingTests: XCTestCase {
...
+    func testWindowHandleSize() throws {
+        let mockWebDriver: MockWebDriver = MockWebDriver()
+        let session = Session(webDriver: mockWebDriver, existingId: "mySession")
+        mockWebDriver.expect(path: "session/mySession/window/myWindow/size", method: .post)
+        try session.resize(window: "myWindow", width: 500, height: 500)
+
+        mockWebDriver.expect(path: "session/mySession/window/myWindow/size", method: .get, type: Requests.SessionWindowSize.Get.self) {
+            ResponseWithValue(.init(width: 500, height: 500))
+        }
+        XCTAssert(try session.size(window: "myWindow") == (width: 500, height: 500))
```
Comment (compnerd) [inline Tests/UnitTests/APIToRequestMappingTests.swift:192]: I think that a nice follow up would be to hang `size` off of `Window` so you could do
```swift
session.window(named: "myWindow").size == (width: 500, height: 500)
```

**Review by compnerd — APPROVED** [review]

**Pull request merged**